src/types.cpp

```diff
@@ -615,11 +615,11 @@ Rational floatToRationalCast(float f) {
   const double d = f;
   // Beware: primitive conversion algorithm
   int32_t den;
-  if (std::fabs(d) <= std::numeric_limits<int32_t>::max() / 1000000) {
+  if (std::fabs(d) <= std::numeric_limits<int32_t>::max() / 1000000.0) {
```

Comment:
I think the integer rounding is deliberate here.
```suggestion
  if (std::fabs(d) <= static_cast<double>(std::numeric_limits<int32_t>::max() / 1000000)) {
```

Reply:
Removed.